### PR TITLE
Add form validation

### PR DESCRIPTION
**Pull request by Szulerinio**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -141,26 +138,41 @@ function App() {
...
             color={
-              numberOfJokesToFetch < 1 || numberOfJokesToFetch > 100
+              downloadJokesFormik.errors.numberOfJokesToFetch
                 ? ButtonColors.Gray
                 : ButtonColors.Dark
             }
             wide
-            disabled={numberOfJokesToFetch < 1 || numberOfJokesToFetch > 100}
-            onClick={() => downloadJokes(numberOfJokesToFetch)}
+            disabled={!!downloadJokesFormik.errors.numberOfJokesToFetch}
```
Comment (Szulerinio) [inline src/App.tsx:161]: Is this conversion to boolean ok?

File: src/App.tsx [hunk 2]
```diff
@@ -22,29 +24,24 @@ function App() {
...
-      return parseInt(value) > 100
-        ? 100
-        : parseInt(value) < 0
-        ? 0
-        : parseInt(value);
-    });
-  };
+  const downloadJokesFormik = useFormik({
+    initialValues: {
+      numberOfJokesToFetch: "0",
```
Comment (radKoz) [inline src/App.tsx:31]: should `0` be in `""` if its a number?

File: src/App.tsx [hunk 3]
```diff
@@ -22,29 +24,24 @@ function App() {
...
+    },
+    validationSchema: Yup.object({
+      numberOfJokesToFetch: Yup.number()
+        .max(100, t("error.tooHighNumber", { number: 100 }))
```
Comment (radKoz) [inline src/App.tsx:35]: you can store `100` in a const so later if it changes you'll change it only in one place

File: src/App.tsx (hunk 1 above)
Comment (radKoz) [reply src/App.tsx:161]: yes

File: src/locales/en/translation.json [hunk 4]
```diff
@@ -4,5 +4,10 @@
   "saveJoke_other": "Save {{count}} Jokes",
```
Comment (radKoz) [inline src/locales/en/translation.json:4]: saveJokeOther, try not to mix different naming conventions
btwy what's the difference between `saveJoke_zero` and `saveJoke_one` if translation is the same?

**Review by radKoz — APPROVED** [review]

File: src/locales/en/translation.json (hunk 4 above)
Comment (Szulerinio) [reply src/locales/en/translation.json:4]: I followed the convention used in [i18next docs](https://www.i18next.com/translation-function/plurals)

I guess that mixing of cases is used to separate "name" part from "count" part and is intentional 

I will try to use Interval plurals for combining _zero and _one

File: src/App.tsx (hunk 2 above)
Comment (Szulerinio) [reply src/App.tsx:31]: storing that as a string makes it easier to handle a case when user deletes all numbers from input before providing any other
If it's a rule that numbers should be stored as numbers i can change it, but i feel like its easier that way in this case

**Pull request merged**